core/utils/misc.hpp

```diff
@@ -107,40 +108,44 @@ move_on_copy<T> make_move_on_copy(T&& value) noexcept {
 /// @brief convenient helper for caching function results
 ////////////////////////////////////////////////////////////////////////////////
 template<
-  typename Input,
-  Input Size,
-  typename Func,
-  typename = typename std::enable_if_t<std::is_integral_v<Input>>
-> class cached_func {
+    typename Input,
+    Input Size,
+    typename Func,
+    typename = typename std::enable_if_t<std::is_integral_v<Input>>>
```

Comment:
When signed?

Reply:
Do you see any issues with signed/unsigned types?

Reply:
Not, you right, I thought if unsigned we can replace `<` to `!=`, but this is not true